include/zephyr/net/coap.h

```diff
@@ -696,6 +696,27 @@ static inline uint16_t coap_block_size_to_bytes(
 	return (1 << (block_size + 4));
 }
 
+/**
+ * @brief Helper for converting block size in bytes to enumeration.
+ *
+ * NOTE: Only valid CoAP block sizes map correctly.
+ *
+ * @param bytes CoAP block size in bytes.
+ * @return enum coap_block_size
+ */
+static inline enum coap_block_size coap_bytes_to_block_size(uint16_t bytes)
```

Comment:
It is required to convert byte values into enum values which match the [RFC 7959 SZX field](https://datatracker.ietf.org/doc/html/rfc7959#section-2.2)

0 equals to 16 bytes.
1 equals to 32 bytes
...
6 equals to 1024 bytes.

Reply:
I see.
I still think that converting between `enum coap_block_size` and number of bytes should be in `coap.h`.
Other direction is already there.

Then LwM2M is just an user of that API. There could be others that use the same.

Reply:
I agree. Then probably it would be good to use this new function inside `lwm2m_default_block_size()` as well.

What about the other suggestion to modify the `block_size` parameter in `build_msg_block_for_send()` to minimise the number of changes and conversions?

Reply:
I'll refactor the `lwm2m_default_block_size()` as well, but for function parameters I would like to keep as enumerations. There is at least a possibility that compiler can detect wrong values that are outside of the enumerations.
If I use just `uint16_t` it is unsafer, because not all values could be used.